logging/logging.go

```diff
@@ -27,33 +36,20 @@ type Logger interface {
 	IsVerbose() bool
 }
 
-// WithInfoErrorWriter is an optional interface for loggers that want to support a separate writer for errors and standard logging.
-// the DebugInfoWriter should write to stderr if quiet is false.
-type WithInfoErrorWriter interface {
-	InfoErrorWriter() io.Writer
+// WithSelectableWriter is an optional interface for loggers that want to support a separate writer per log level.
+type WithSelectableWriter interface {
```

Comment:
Condensed two interfaces into one to increase robustness and in my opinion provide additional clarity. e.g. `GetInfoErrorWriter` was confusing.